opportunity_job_app/src/api/baseApi.ts

```diff
@@ -3,7 +3,7 @@
 import axios from "axios";
 import { AuthService } from "./authService";
 
-const api = axios.create({
+export const api = axios.create({
```

Comment:
`api` ne bi trebalo exportovati, vec bi trebalo koristiti `API` objekat koji se vec eksportuje na kraju ovog fajla.
Sve potrebne request-ove saljes pomocu `API` objekta, tj. pozivas funkcije nad `API` objektom gde god ti trebaju.
Mozda bi bilo bolje `api` preimenovati u `axiosInstance` ili tome slicno da ne buni s obzirom da imamo i `API`. 

Sto se tice samih funkcija u `API` objektu, njih bi refaktorisao tako da svaka koristi auth headers, bez obzira da li BE API zahtevao to ili ne (za odredjeni endpoint).
Treba proveriti da li je bi ovo predstavljalo problem za endpointe koji ne zahtevaju auth headers.

Tako da bi `API` objekat izgledao ovako nekako:
```
const API = {
    getResourceList: (resource: string, queryParams = "") => {
        const config = { headers: AuthService.getAuthorizationHeaders() };
        return api.get(`${ resource }/?${ queryParams }`, config);
    },
    getResourceDetails: (resource: string, resourceId: number) => {
        const config = { headers: AuthService.getAuthorizationHeaders() };
        return api.get(`${ resource }/${ resourceId }/`, config);
    },
    createResource: (url: string, data: Object) => {
        const config = { headers: AuthService.getAuthorizationHeaders() };
        return api.post(url, data, config);
    },
    patchResource: (url: string, data: Object) => {
        const config = { headers: AuthService.getAuthorizationHeaders() };
        return api.patch(url, data);
    },
};
```

Iz naziva funkcija je izbacena `protected` rec jer su sve funkcije imaju auth headers kojeg su prethodno imale samo "protected" funkcije.